### ISSUE TITLE
semgrep reports vulnerabilities in generated mkdocs site

### ISSUE THREAD

**Issue by jorgheymans** [squidfunk/mkdocs-material#3052]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

For a basic generated mkdocs site with material, semgrep reports below issue on the 'made with' link for each page : 

```
html.security.missing-noreferrer.missing-noreferrer
     > public/main/404.html:233
     ╷
  233│   <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
     ╵
     = This anchor tag with 'target="_blank"' is missing 'noreferrer'. A page
       opened with 'target="_blank"' can access the window object of the origin
       page. This means it can manipulate the 'window.opener' property, which
       could redirect the origin page to a malicious URL. This is called reverse
       tabnabbing. To prevent this, include 'rel=noreferrer' on this tag.
html.security.missing-noreferrer.missing-noreferrer
     > public/main/index.html:244
     ╷
  244│   <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
     ╵
     = This anchor tag with 'target="_blank"' is missing 'noreferrer'. A page
       opened with 'target="_blank"' can access the window object of the origin
       page. This means it can manipulate the 'window.opener' property, which
       could redirect the origin page to a malicious URL. This is called reverse
       tabnabbing. To prevent this, include 'rel=noreferrer' on this tag.
```

They don't really pose a security threat for our purposes, but semgrep is finding them and it's hard to exclude that particular rule just for mkdocs' sake. So i'm hoping it's an easy fix that could be applied to the theme itself. 

The ruleset we applied are  these:
```
      p/security-audit
```

### Expected behaviour

semgrep does not report security vulnerabilities

### Actual behaviour

security issues exist

### Steps to reproduce

$ mkdocs build
$ cd site
$ semgrep --config=p/security-audit

### Package versions

$ python --version
Python 3.9.7
$ mkdocs --version
mkdocs, version 1.2.2 from /usr/local/lib/python3.9/site-packages/mkdocs (Python 3.9)
$ pip show mkdocs-material | grep -E ^Version
Version: 7.3.0

### Configuration

```yaml
site_name: mkdocs
site_url: http://localhost:8000/docs
docs_dir: mkdocs
theme:
  name: material
  palette:
    scheme: preference
    primary: teal
    accent: amber
  language: en
```

### System information

- Operating system: Ubuntu
- Browser: Firefox

**Comment by jorgheymans** [comment]
As a workaround i noticed the banner can be disabled :

```
  extra:
    generator: false
```

**Comment by squidfunk** [comment]
Thanks for reporting.

> They don't really pose a security threat for our purposes, but semgrep is finding them and it's hard to exclude that particular rule just for mkdocs' sake. So i'm hoping it's an easy fix that could be applied to the theme itself.

I understand that you're having trouble with this tool (which I never heard of by the way), but a referrer is crucial to understand where traffic is coming from. This is essential information for a solid growth and marketing strategy, which is essential for a successful Open Source project.

However, it's an easy fix: 7.3.0 (released yesterday) added the ability to [remove the 'Made with Material for MkDocs' notice](https://squidfunk.github.io/mkdocs-material/setup/setting-up-the-footer/#remove-generator), also getting rid of the link.
